Source/SwiftLintFramework/Reporters/JUnitReporter.swift

```diff
@@ -10,17 +10,30 @@ public struct JUnitReporter: Reporter {
     }
 
     public static func generateReport(_ violations: [StyleViolation]) -> String {
-        return "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n<testsuites><testsuite>" +
-            violations.map({ violation -> String in
-                let fileName = (violation.location.file ?? "<nopath>").escapedForXML()
-                let severity = violation.severity.rawValue + ":\n"
-                let message = severity + "Line:" + String(violation.location.line ?? 0) + " "
-                let reason = violation.reason.escapedForXML()
-                return [
-                    "\n\t<testcase classname='Formatting Test' name='\(fileName)\'>\n",
-                    "<failure message='\(reason)\'>" + message + "</failure>",
-                    "\t</testcase>"
-                ].joined()
-            }).joined() + "\n</testsuite></testsuites>"
+        let warningCount = violations.filter({ $0.severity == .warning }).count
+        let errorCount = violations.filter({ $0.severity == .error }).count
+
+        return [
+            "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n",
+            "<testsuites failures=\"\(warningCount)\" errors=\"\(errorCount)\">\n",
+            "\t<testsuite failures=\"\(warningCount)\" errors=\"\(errorCount)\">",
+            violations.map({ testCase(for: $0) }).joined(),
+            "\n\t</testsuite>\n",
+            "</testsuites>"
+        ].joined()
```

Comment:
Can we use a text block instead?

Reply:
Would be also possible, but I did it this way, because `CheckstyleReporter`, `GitHubActionsLoggingReporter`, `GitLabJUnitReporter`, `HTMLReporter` and `XcodeReporter` also used this style.
Should I change it anyway?

Reply:
Fine for now. Let me refactor them in a follow-up.